store/tikv/region_request.go

```diff
@@ -55,109 +55,386 @@ func NewRegionRequestSender(bo *Backoffer, regionCache *RegionCache, client Clie
 	}
 }
 
-// SendKVReq sends a KV request to tikv server.
-func (s *RegionRequestSender) SendKVReq(req *kvrpcpb.Request, regionID RegionVerID, timeout time.Duration) (*kvrpcpb.Response, error) {
+// RegionErrorResponse is an interface used for retry on region errors in
+// the implemetation of RegionRequestSender.
+type RegionErrorResponse interface {
+	GetRegionError() *errorpb.Error
+}
+
+type rpcFunc func(ctx *RPCContext) (RegionErrorResponse, error, bool)
```

Comment:
Will it be better to put the returned error at the last position? cc @tiancaiamao